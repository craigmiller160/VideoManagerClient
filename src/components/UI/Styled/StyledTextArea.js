import styled from 'styled-components';

const StyledTextArea = styled.textarea`
    resize: none;
    display: block;
    border: 1px solid lightgray;
    border-radius: 3px;
    width: 100%;
`;
StyledTextArea.displayName = 'StyledTextArea';

export default StyledTextArea;