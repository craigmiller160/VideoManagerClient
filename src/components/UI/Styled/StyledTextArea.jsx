/*
 *     VideoManagerClient
 *     Copyright (C) 2020 Craig Miller
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledTextArea = styled.textarea`
	display: block;
	border: 1px solid ${(props) => (props.hasError ? '#dc3545' : 'lightgray')};
	border-radius: 3px;
	width: 100%;
	resize: ${(props) => (props.resize ? 'both' : 'none')};
`;
StyledTextArea.displayName = 'StyledTextArea';
StyledTextArea.propTypes = {
	resize: PropTypes.bool,
	hasError: PropTypes.bool
};
StyledTextArea.defaultProps = {
	resize: false,
	hasError: false
};

export default StyledTextArea;
