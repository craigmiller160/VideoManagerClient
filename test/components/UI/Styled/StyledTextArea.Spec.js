import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components'
import StyledTextArea from 'components/UI/Styled/StyledLabel';

describe('StyledTextArea', () => {
    it('renders correctly', () => {
        const component = mount(<StyledTextArea/>);
        expect(component).toMatchSnapshot();
    });
});