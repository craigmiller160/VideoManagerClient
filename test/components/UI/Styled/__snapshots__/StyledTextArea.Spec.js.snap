// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextArea renders correctly 1`] = `
.c0 {
  font-weight: bold;
  margin: 0;
}

<StyledLabel>
  <label
    className="c0"
  />
</StyledLabel>
`;
